Dockerfile.ci

```diff
@@ -1181,7 +1181,7 @@ ARG PYTHON_BASE_IMAGE
 ARG AIRFLOW_IMAGE_REPOSITORY="https://github.com/apache/airflow"
 
 # By increasing this number we can do force build of all dependencies
-ARG DEPENDENCIES_EPOCH_NUMBER="11"
+ARG DEPENDENCIES_EPOCH_NUMBER="12"
```

Comment:
Actually it seems that it does not work this way. The sqlalchemy-redshift connector is still installed in our "main" caching step.

You can check it by looking at the constraints generated as artifacts in this build - sqlalchemy-redshift connector is still there.

The reason is, because we are using caching from the `main` installation for this PR - and in main, the dependency is still there. I think I will need to add a new label that you should be able to set for such PR to not use cache during the image build. 

In breeze this is done with `--docker-cache disabled` and in CI this is triggered when `disable image cache` label is set. I will apply the label and re-run it. If it will work, I will update the docs/comments to make it clear that you need to do it as well.

Reply:
(BTW. Increasing EPOCH in this case is still needed - because local breeze images will still keep the old cached layer even after we merge this change).

Reply:
Actually It turned out that CI image build did not handle cache disabling properly. https://github.com/apache/airflow/pull/42967 should be merged and this one should be rebased on top of it after that in order to properly test package removal.

Such build will be much slower than the cached build, but it will guarantee that the dependencies are not installed in a pr that has "docker cache disabled" label set - neither from image cache  in ghcr.io nor from "main" repository.